File: Python/LeetCode/739.py
class Solution(object):
    def dailyTemperatures(self, T):
        """
        :type T: List[int]
        :rtype: List[int]
        """
        stack = []
        result = [0]*len(T)

        for i in range(len(T)):
            while stack and T[stack[-1]] < T[i]:
                index = stack.pop()
                result[index] = i - index
            stack.append(i)

        return result

File: Cpp/LeetCode/739.cpp
//
// Created by caizixiang on 2019-07-30.
//

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        vector<int> result(T.size(), 0);
        vector<int> stack;

        for (int i = 0; i < T.size(); i++)
        {
            while (!stack.empty() && T[stack.back()] < T[i])
            {
                int index = stack.back();
                stack.pop_back();
                result[index] = i - index;
            }
            stack.push_back(i);
        }

        return result;
    }
};